Sources/NIOHTTPCompression/HTTPCompression.swift

```diff
@@ -0,0 +1,130 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import CNIOExtrasZlib
+import NIO
+
+public enum HTTPCompression {
+
+    public enum CompressionAlgorithm: String {
+        case gzip
+        case deflate
+    }
+    
+    public enum CompressionError: Error {
+        case uncompressedWritesPending
```

Comment:
I must say converting this to a struct makes catching this error a little more awkward. Instead of
```
catch NIOHTTPCompressionSettings.CompressionError.uncompressedWritesPending
```
You have to replace it with 
```
catch let error as NIOHTTPCompressionSettings.CompressionError:
    if error == .uncompressedWritesPending {
         ...
```

Reply:
It makes catching it awkward, but if we don't do it we can never add another error to this enum: we have to mint new enums every time. It's a bad scene.

Reply:
done

Reply:
@adam-fowler yeah, the `switch` is unfortunate. In [this document](https://gist.github.com/weissi/d861bed9cad17871c3ad814911feeafb), I explored an alternative: 

```
} catch let error as NIOHTTPCompressionError where error == .uncompressedWritesPending {
```

not much nicer but...